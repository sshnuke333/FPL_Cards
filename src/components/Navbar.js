import React from 'react';
import {
    StyledBanner,
    StyledNav,
    Logo,
    MenuButton,
    NavList,
    NavItem,
    StyledLink,
} from './Navbar.styles';

import { toggleDisplay } from '../helpers/display';

export const Navbar = () => {
    return (
        <StyledBanner>
            <StyledNav>
                <StyledLink
                    to="/"
                    aria-label="FPL Cards Home Page on logo link"
                >
                    <Logo alt="FPL Cards logo" />
                </StyledLink>
                <MenuButton
                    id="menu-button"
                    aria-expanded="false"
                    onClick={(e) => toggleDisplay(e, 'menu-list', 'flex')}
                >
                    Menu
                </MenuButton>
                <NavList id="menu-list" role="menu">
                    <NavItem>
                        <StyledLink to="/">Home</StyledLink>
                    </NavItem>
                    <NavItem>
                        <StyledLink to="play">Play</StyledLink>
                    </NavItem>
                    <NavItem>
                        <StyledLink to="disclaimer">Disclaimer</StyledLink>
                    </NavItem>
                </NavList>
            </StyledNav>
        </StyledBanner>
    );
};
