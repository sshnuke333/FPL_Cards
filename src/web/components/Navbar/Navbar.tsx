import React from 'react';
import {
    StyledBanner,
    StyledNav,
    Logo,
    MenuButton,
    NavList,
    NavItem,
    StyledLink,
} from './Navbar.styles';

import { expandTarget } from '../../helpers/display';

const Navbar = (): JSX.Element => (
    <StyledBanner>
        <StyledNav>
            <StyledLink to="/" aria-label="FPL Cards Home Page on logo link">
                <Logo alt="FPL Cards logo" />
            </StyledLink>
            <MenuButton
                id="menu-button"
                aria-expanded="false"
                aria-controls="menu-list"
                onClick={(e) => expandTarget(e, 'menu-list', 'flex')}
            >
                Menu
            </MenuButton>
            <NavList id="menu-list" role="menu">
                <NavItem>
                    <StyledLink to="/">Home</StyledLink>
                </NavItem>
                <NavItem>
                    <StyledLink to="play">Play</StyledLink>
                </NavItem>
                <NavItem>
                    <StyledLink to="disclaimer">Disclaimer</StyledLink>
                </NavItem>
            </NavList>
        </StyledNav>
    </StyledBanner>
);

export default Navbar;
