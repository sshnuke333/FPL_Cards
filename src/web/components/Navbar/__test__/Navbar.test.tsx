import { render, screen, within } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import { BrowserRouter as Router } from 'react-router-dom';

import Navbar from '../Navbar';

const renderNavbar = (
    <Router>
        <Navbar />
    </Router>
);

describe('Navbar renders', () => {
    it('correctly', () => {
        const { asFragment } = render(renderNavbar);
        expect(asFragment()).toMatchSnapshot();
    });

    beforeEach(() => {
        render(renderNavbar);
    });

    it('a header element', () => {
        expect(screen.getByRole('banner')).toBeInTheDocument();
    });
    it('a navbar element', () => {
        expect(screen.getByRole('navigation')).toBeInTheDocument();
    });
    describe('inside navbar', () => {
        it('a clickable logo', () => {
            const logo = screen.getByRole('link', {
                name: /fpl\scards\shome\s/i,
            });
            expect(logo).toBeInTheDocument();
            expect(within(logo).getByRole('img')).toBeInTheDocument();
        });
        it('a hidden expandable menu botton', () => {
            const button = screen.getByText(/Menu/i);
            expect(button).toBeInTheDocument();
            expect(button).not.toBeVisible();
        });
        it('a menu list', () => {
            expect(screen.getByRole('menu')).toBeInTheDocument();
        });
        it('a Home link in menu', () => {
            const menu = screen.getByRole('menu');
            expect(
                within(menu).getByRole('link', { name: /Home/i })
            ).toBeInTheDocument();
        });
        it('a Play link in menu', () => {
            const menu = screen.getByRole('menu');
            expect(
                within(menu).getByRole('link', { name: /Play/ })
            ).toBeInTheDocument();
        });
        it('a Disclaimer link menu', () => {
            const menu = screen.getByRole('menu');
            expect(
                within(menu).getByRole('link', { name: /Disclaimer/ })
            ).toBeInTheDocument();
        });
    });
});

describe('Menu button', () => {
    beforeEach(() => {
        render(renderNavbar);
    });
    it('is collapsed on render', () => {
        const button = screen.getByText(/Menu/i);
        expect(button).toHaveAttribute('aria-expanded', 'false');
    });
    it('expands and displays navbar list after clicking', () => {
        const button = screen.getByText(/Menu/i);
        const list = screen.getByRole('menu');
        userEvent.click(button);
        expect(button).toHaveAttribute('aria-expanded', 'true');
        expect(list).toBeVisible();
    });
    it('collapses and hides navbar list on clicking again', () => {
        const button = screen.getByText(/Menu/i);
        const list = screen.getByRole('menu');
        userEvent.dblClick(button);
        expect(button).toHaveAttribute('aria-expanded', 'false');
        expect(list).not.toBeVisible();
    });
});
