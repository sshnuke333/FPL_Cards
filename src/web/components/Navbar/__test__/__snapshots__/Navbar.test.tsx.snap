// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar renders correctly 1`] = `
<DocumentFragment>
  <header
    class="sc-bcXHqe PGcvk"
  >
    <nav
      class="sc-gswNZR ACgTW"
    >
      <a
        aria-label="FPL Cards Home Page on logo link"
        class="sc-gKPRtg jiKIul"
        href="/"
      >
        <img
          alt="FPL Cards logo"
          class="sc-dkrFOg kjAtEz"
        />
      </a>
      <button
        aria-controls="menu-list"
        aria-expanded="false"
        class="sc-hLBbgP iUHvhY"
        id="menu-button"
      >
        Menu
      </button>
      <ul
        class="sc-eDvSVe igyxqk"
        id="menu-list"
        role="menu"
      >
        <li
          class="sc-jSUZER dMcPVC"
        >
          <a
            class="sc-gKPRtg jiKIul"
            href="/"
          >
            Home
          </a>
        </li>
        <li
          class="sc-jSUZER dMcPVC"
        >
          <a
            class="sc-gKPRtg jiKIul"
            href="/play"
          >
            Play
          </a>
        </li>
        <li
          class="sc-jSUZER dMcPVC"
        >
          <a
            class="sc-gKPRtg jiKIul"
            href="/disclaimer"
          >
            Disclaimer
          </a>
        </li>
      </ul>
    </nav>
  </header>
</DocumentFragment>
`;
